proto/meta.proto

```diff
@@ -193,6 +193,64 @@ message MetaLeaseInfo {
   uint64 lease_expire_time = 3;
 }
 
+// Below for scale service.
+enum TaskStatus {
+  NOT_FOUND = 0;
+  PENDING = 1;
+  BUILDING = 2;
+  CANCELLED = 3;
+  FINISHED = 4;
+  FAILED = 5;
+}
+
+message ScaleTask {
+  enum TaskType {
+    INVALID = 0;
+    SCALE_IN = 1;
+    SCALE_OUT = 2;
+  }
```

Comment:
I'm not sure whether we should distinguish between scale-in and scale-out tasks. Take the actor migration as an example; it can be formalized as an atomical scale-in and scale-out in a single barrier.
A possible design might be making the `map<uint32, uint32> fragment_parallelism = 5` more general to allow specifying a type(in/out) for every parallel unit of fragments.

cc @yezizp2012 Any ideas? 🥵

Reply:
Here is a propose, we can change like this and don't need task type and any depended host info, this requires operator to bring forward deploying new CNs if needed.
```
message FragmentScalePlan {
  uint32 fragment_id = 1;
  // New parallelism.
  uint32 parallelism = 2;
  // Actor placement: worker id for each actor, its length should equal to parallelism. The parallelism decides whether
  // this is a scale-in or scale-out task, and also this is compatible with simple migration if needed in the future.
  repeated uint32 actor_placement = 3;
}

message ScaleTask {
  String task_id = 1;
  TaskStatus task_status = 2;
  map<string, FragmentScalePlan> scale_plan = 3;
}
```